### ISSUE TITLE
How to use the other optimizer like "Adam" in DefaultTrainer ?

### ISSUE THREAD

**Issue by TriplePool** [facebookresearch/detectron2#668]
Thank you for excellent work !
Is the learning rate reassigned by lr_scheduler every iteration in DefaultTrainer?
If so, should I create a new Trainer ?

**Comment by ppwwyyxx** [comment]
https://detectron2.readthedocs.io/modules/engine.html#detectron2.engine.defaults.DefaultTrainer.build_optimizer

lr_scheduler's `step` is called every step.
